### ISSUE TITLE
How to smartly/early stop model.fit?

### ISSUE THREAD

**Issue by UnixJunkie** [deepchem/deepchem#1533]
Hello,

While playing with this tutorial:
https://deepchem.io/docs/notebooks/graph_convolutional_networks_for_tox21.html

Rather than using this:
```
model.fit(train_dataset, nb_epoch=50)
```
Is it possible to say: "stop using more epochs as soon as the AUC on the validation set doesn't improve by more than 0.0001"?

I mean, nb_epoch should be a maximum limit.
If the model doesn't improve enough way before that limit, I think training should be stopped and my precious time saved.

Regards,
F.

**Comment by UnixJunkie** [comment]
I'll play with the restore parameter.
Looks like it was made for this.

**Comment by peastman** [comment]
Something like this should do what you want.

```python
for i in range(5):
  model.fit(train_dataset, nb_epoch=10)
  if (whatever logic you want here):
    break
```
